### PR TITLE
Fix a bug that if a node changes parent, parent's matrix will not update to reflect this change.

### PR DESCRIPTION
**Pull request by darkdukey**
1. Create a test case Bug-Child to reproduce this bug
2. Fixes it at `insertChild()`

### CODE REVIEW

File: tests/cpp-tests/Classes/BugsTest/Bug-Child.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+bool BugChild::init()
+{
+    if (BugsTestBaseLayer::init())
+    {
+        auto size = Director::getInstance()->getWinSize();
+        
+        // create and initialize a Label
+        auto item1 = MenuItemFont::create("Switch Child", CC_CALLBACK_1(BugChild::switchChild, this));
+        
+        menu = Menu::create(item1, NULL);
```
Comment (minggo) [inline tests/cpp-tests/Classes/BugsTest/Bug-Child.cpp:34]: NULL -> nullptr

**Comment by minggo** [comment]
Thanks.

**Comment by WenhaiLin** [comment]
@darkdukey I submit another PR at  https://github.com/cocos2d/cocos2d-x/pull/6714  base on your PR to update project for other platforms.

**Comment by darkdukey** [comment]
Thanks, you can close this one.

**Pull request closed without merging**